### ISSUE TITLE
Toggling readOnly prop of SandpackCodeEditor breaks editor

### ISSUE THREAD

**Issue by anthonygore** [codesandbox/sandpack#454]
# Bug report

## Packages affected

-   [ ] sandpack-client
-   [x] sandpack-react

## Description of the problem

When the `readOnly` prop of `SandpackCodeEditor` is toggled from false to true and then back to false again the editor functionality and syntax highlighting is broken.

### What steps can we take to reproduce the problem?

```jsx
const [readOnly,setReadOnly] = useState(false)
const handleClick = () => {
  setReadOnly(() => true)
}
return <SandpackCodeEditor readOnly={readOnly} />
```

Before:

![Screen Shot 2022-05-02 at 1 58 43 PM](https://user-images.githubusercontent.com/5368448/166182654-c2ebcac5-6adb-4637-a215-7ce701e9ff17.png)

After value is toggled twice:

![Screen Shot 2022-05-02 at 1 59 07 PM](https://user-images.githubusercontent.com/5368448/166182647-2622f0b6-5b84-4942-b8f0-19c3c075cc5e.png)

### Your Environment

| Software                | Name/Version |
| ----------------------- | ------------ |
| Sandpack-client version |              |
| Sandpack-react version  |      0.19.4        |
| Browser                 |       Chrome       |
| Operating System        |       MacOS       |

**Comment by DeMoorJasper** [comment]
Can reproduce this, sandbox to reproduce this: https://codesandbox.io/s/sparkling-tdd-gqvm5z?file=/src/App.tsx

**Comment by danilowoz** [comment]
Thanks for reporting it, it has been fixed on #456

**Comment by anthonygore** [comment]
Thanks!
